libconcurrent/primitives/threadtools.c

```diff
@@ -191,7 +256,9 @@ inline int32_t synchGetPosixThreadId(void) {
 }
 
 inline void synchResched(void) {
-    if (__noop_resched) {
+    if (__noop_resched && synchGetMachineModel() != INTEL_X86_MACHINE) {
```

Comment:
this check already exists in `synchPause`, I guess you have it here to save a function call, right?

Reply:
Cleaned-up. Check a97eef.